cmd/segment-reaper/observer.go

```diff
@@ -127,6 +146,12 @@ func (obsvr *observer) processSegment(ctx context.Context, path metainfo.ScopedP
 		}
 	}
 
+	if obsvr.from != nil && obsvr.from.Before(pointer.CreationDate) {
+		object.skip = true
+	} else if obsvr.to != nil && obsvr.to.After(pointer.CreationDate) {
+		object.skip = true
+	}
```

Comment:
Are the conditions here correct?

Let's say `from` is Dec-01, `to` is Dec-09, and `pointer.CreationDate` is Dec-04.

I would expect that this object will be analyzed because its `pointer.CreationDate` is between `from` and `to`. But reading the code, it seems that it will be skipped: first, because `obsrv.from` (Dec-01) is _before_ `pointer.CreationDate` (Dec-04); and second, because `obsrv.to` (Dec-09) is _after_ `pointer.CreationDate` (Dec-04)

Reply:
good catch, this definitely needs unit test